app/javascript/v3/helpers/specs/BrowserHelper.spec.js

```diff
@@ -0,0 +1,43 @@
+import { getBrowserLocale, getBrowserTimezone } from '../BrowserHelper';
+
+describe('getBrowserLocale', () => {
+  it('should return the correct locale code when an exact match is found', () => {
+    // Mock window.navigator.language
+    Object.defineProperty(window.navigator, 'language', {
+      value: 'en-US',
+      configurable: true,
+    });
+
+    const languages = [{ iso_639_1_code: 'en' }, { iso_639_1_code: 'en_US' }];
+    expect(getBrowserLocale(languages)).toBe('en');
+  });
+
+  it('should return the correct locale code when only a partial match is found', () => {
+    Object.defineProperty(window.navigator, 'language', {
+      value: 'en-GB',
+      configurable: true,
+    });
+
+    const languages = [{ iso_639_1_code: 'en' }, { iso_639_1_code: 'fr' }];
+    expect(getBrowserLocale(languages)).toBe('en');
+  });
+
+  it('should return undefined when no match is found', () => {
+    Object.defineProperty(window.navigator, 'language', {
+      value: 'es-ES',
+      configurable: true,
+    });
+
+    const languages = [{ iso_639_1_code: 'en' }, { iso_639_1_code: 'fr' }];
+    expect(getBrowserLocale(languages)).toBeUndefined();
+  });
+});
```

Comment:
I believe this change would leak to other tests. You need to use something like `jest.spyOn`.